app/main.go

```
@@ -1 +1,122 @@
 package main
+
+import (
+	"fmt"
+	"html/template"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	flags "github.com/jessevdk/go-flags"
+
+	"github.com/akellbl4/wod/app/config"
+	"github.com/akellbl4/wod/app/domain"
+)
+
+// Opts with all cli commands and flags
+type Opts struct {
+	ConfigPath      string `long:"config" short:"c" env:"WOD_CONFIG" description:"Path to config.json" default:"./config.json"`
+	TemplatePath    string `long:"template" short:"t" env:"WOD_TEMPLATE_PATH" description:"Path to html template" default:"./template.html.tmpl"`
+	DestinationPath string `long:"destination" short:"d" env:"WOD_DESTINATION" description:"Path to destination directory" default:"./web"`
+}
+// TemplateData with all template variables
+type TemplateData struct {
+	Domain           string
+	Price            int
+	RegistrationDate time.Time
+}
+
+func main() {
+	var opts Opts
+
+	if _, err := flags.Parse(&opts); err != nil {
+		os.Exit(1)
+	}
+
+	cfg, err := config.Parse(opts.ConfigPath)
+
+	if err != nil {
+		os.Exit(1)
+	}
+
+	tmpl := template.Must(template.ParseFiles(opts.TemplatePath))
+	err = creationDestinationFolder(opts.DestinationPath)
+
+	if err != nil {
+		log.Printf("Error on creation destination folder")
+		os.Exit(1)
+	}
+
+	err = createPages(cfg, tmpl, opts.DestinationPath)
+
+	if err != nil {
+		os.Exit(1)
+	}
+}
+
+func creationDestinationFolder(path string) error {
+	_, err := os.Stat(path)
+
+	if os.IsNotExist(err) {
+		os.Mkdir(path, os.ModePerm)
```

error is not checked